roachtest: run INSPECT as background job and poll for completion

spilchen opened the pull request:
Long-running INSPECT statements (e.g., >3h) have previously failed due to connection resets, despite no server crash. This likely stems from a missed keep-alive or a client timeout, though the exact cause is unclear.

This change updates the roachtest to:
- Run the INSPECT statement as a background job using a short statement_timeout (5s).
- Poll the job table for job completion instead of waiting synchronously.
- Report progress at 10% intervals to improve visibility without overwhelming the logs.

Fixes #155610

Release note: none

Epic: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/155653)
<!-- Reviewable:end -->

bghal reviewed (approved):
> @bghal reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/155653)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)

___
*[`pkg/cmd/roachtest/tests/inspect_throughput.go` line 304 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/155653#-Oc671ph42dg8gXmZWrK:-Oc671ph42dg8gXmZWrL:bppdyp3) ([raw file](https://github.com/cockroachdb/cockroach/blob/a2a23a27493ff42cbb3038057effe2fccdaa2732/pkg/cmd/roachtest/tests/inspect_throughput.go#L304)):*
> ```Go
> ) (jobID int64) {
> 	// Set a short statement timeout to force INSPECT to run as a background job.
> 	if _, err := db.Exec("SET statement_timeout = '5s'"); err != nil {
> ```

Nit: Seems like the `require` package can be used in roachtests.
___
*[`pkg/cmd/roachtest/tests/inspect_throughput.go` line 311 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/155653#-Oc69Cpd9KU8ak_1KUt1:-Oc69Cpd9KU8ak_1KUt2:b-tpbqc2) ([raw file](https://github.com/cockroachdb/cockroach/blob/a2a23a27493ff42cbb3038057effe2fccdaa2732/pkg/cmd/roachtest/tests/inspect_throughput.go#L311)):*
> ```Go
> 
> 	// Reset statement timeout.
> 	if _, resetErr := db.Exec("RESET statement_timeout"); resetErr != nil {
> ```

Does this need to break up the error checking? Could it be deferred?
___
*[`pkg/cmd/roachtest/tests/inspect_throughput.go` line 316 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/155653#-Oc69Q5w9rRiM6rYpvjB:-Oc69Q5w9rRiM6rYpvjC:b-jmlpxd) ([raw file](https://github.com/cockroachdb/cockroach/blob/a2a23a27493ff42cbb3038057effe2fccdaa2732/pkg/cmd/roachtest/tests/inspect_throughput.go#L316)):*
> ```Go
> 
> 	// We expect a statement timeout error, which is normal.
> 	if err != nil && !strings.Contains(err.Error(), "statement timeout") {
> ```

Nit: Similarly `require.ErrorContains` would be nice.
___
*[`pkg/cmd/roachtest/tests/inspect_throughput.go` line 361 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/155653#-Oc6BVqGBBQWSFOEWgWN:-Oc6BVqGBBQWSFOEWgWO:b-6ejf7d) ([raw file](https://github.com/cockroachdb/cockroach/blob/a2a23a27493ff42cbb3038057effe2fccdaa2732/pkg/cmd/roachtest/tests/inspect_throughput.go#L361)):*
> ```Go
> 
> 			// Check if job is complete.
> 			if status == "succeeded" {
> ```

Nit: Perhaps a switch on `jobs.StatusMessage`?


<!-- Sent from Reviewable.io -->

spilchen committed:
roachtest: run INSPECT as background job and poll for completion

Long-running INSPECT statements (e.g., >3h) have previously failed due
to connection resets, despite no server crash. This likely stems from a
missed keep-alive or a client timeout, though the exact cause is
unclear.

This change updates the roachtest to:
- Run the INSPECT statement as a background job using a short
  statement_timeout (5s).
- Poll the job table for job completion instead of waiting
  synchronously.
- Report progress at 10% intervals to improve visibility without
  overwhelming the logs.

Fixes #155610

Release note: none

Epic: none

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/155653)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @bghal)

___
*[`pkg/cmd/roachtest/tests/inspect_throughput.go` line 304 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/155653#-Oc671ph42dg8gXmZWrK:-Oc6hhCJ59tx_WJdBIqe:b-ovwaw7) ([raw file](https://github.com/cockroachdb/cockroach/blob/a2a23a27493ff42cbb3038057effe2fccdaa2732/pkg/cmd/roachtest/tests/inspect_throughput.go#L304)):*
<details><summary><i>Previously, bghal wrote…</i></summary><blockquote>

Nit: Seems like the `require` package can be used in roachtests.
</blockquote></details>

Good idea.
___
*[`pkg/cmd/roachtest/tests/inspect_throughput.go` line 311 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/155653#-Oc69Cpd9KU8ak_1KUt1:-Oc6huSvE-FKM2g-CeH0:btbxsx4) ([raw file](https://github.com/cockroachdb/cockroach/blob/a2a23a27493ff42cbb3038057effe2fccdaa2732/pkg/cmd/roachtest/tests/inspect_throughput.go#L311)):*
<details><summary><i>Previously, bghal wrote…</i></summary><blockquote>

Does this need to break up the error checking? Could it be deferred?
</blockquote></details>

I'll reset this in a `defer`
___
*[`pkg/cmd/roachtest/tests/inspect_throughput.go` line 361 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/155653#-Oc6BVqGBBQWSFOEWgWN:-Oc6mbsS7m0lZvb-Pl6v:b-ovwaw7) ([raw file](https://github.com/cockroachdb/cockroach/blob/a2a23a27493ff42cbb3038057effe2fccdaa2732/pkg/cmd/roachtest/tests/inspect_throughput.go#L361)):*
<details><summary><i>Previously, bghal wrote…</i></summary><blockquote>

Nit: Perhaps a switch on `jobs.StatusMessage`?
</blockquote></details>

Good idea.


<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR!

bors r+
